deployments/common/image/common-scripts/env-compile

```diff
@@ -59,7 +60,11 @@ if [[ "$USE_FROZEN" != "1" ]]; then
     for pip in ${all_constraints}; do
         echo ${pip}   # show one per line.
     done
-    rm -f /opt/environments/${env}/requirements.txt
-    pip-compile  -v  --output-file /opt/environments/${env}/requirements.txt  --pip-args "${PIP_SWITCHES}"  ${all_constraints}
+
+    # If pip-compile finds an existing requirements.txt file that fulfils the
+    # dependencies then no changes will be made, even if updates are available.
```

Comment:
I'm confused about this.  If USER_FROZEN=0, then don't we want to get the updates?

Reply:
Negative logic was a poor choice there but it works.   The idea is that if USE_FROZEN=0,  (USE_FROZEN != 1) we're determining fully pinned requirements and should re-compile to update.   Otherwise the frozen requirements remain unchanged.  In either case,  the pip-sync command later installs what you ask for,  either newly updated or frozen and unchanged.

Reply:
The pip-compile related lines changed (look deleted) as well as moved down due to the addition of ${env_reqs}.

Reply:
Ah, I see, I didn't read it closely enough.  Thanks.